### PR TITLE
Optional deterministic index in RPC account_create

### PR DESCRIPTION
**Pull request by guilhermelawless**
Adds optional `index` parameter to RPC [account_create](https://github.com/nanocurrency/nano-node/wiki/RPC-protocol#account-create). Wiki will need to be changed.

Additionally extends `account_create` unit tests.

### CODE REVIEW

**Comment by argakiig** [comment]
```
diff --git a/nano/core_test/rpc.cpp b/nano/core_test/rpc.cpp
index 9fb6fc9..ab550b4 100644
--- a/nano/core_test/rpc.cpp
+++ b/nano/core_test/rpc.cpp
@@ -147,7 +147,7 @@ TEST (rpc, account_create)
 	nano::uint256_union account1;
 	ASSERT_FALSE (account1.decode_account (account_text1));
 	ASSERT_TRUE (system.wallet (0)->exists (account1));
-	request.put ("index", max_index+1);
+	request.put ("index", max_index + 1);
 	test_response response2 (request, rpc, system.io_ctx);
 	system.deadline_set (5s);
 	while (response2.status == 0)
diff --git a/nano/node/rpc.cpp b/nano/node/rpc.cpp
index 098a1cd..8f5412a 100644
--- a/nano/node/rpc.cpp
+++ b/nano/node/rpc.cpp
@@ -429,16 +429,16 @@ void nano::rpc_handler::account_create ()
 		const bool generate_work = request.get<bool> ("work", true);
 		nano::account new_key;
 		auto index_text (request.get_optional<std::string> ("index"));
-		if (index_text.is_initialized())
+		if (index_text.is_initialized ())
 		{
 			uint64_t index;
-			if (decode_unsigned (index_text.get(), index) || index > (uint64_t) std::numeric_limits<uint32_t>::max ())
+			if (decode_unsigned (index_text.get (), index) || index > (uint64_t)std::numeric_limits<uint32_t>::max ())
 			{
 				ec = nano::error_common::invalid_index;
 			}
 			else
 			{
-				new_key = wallet->deterministic_insert ((uint32_t) index, generate_work);
+				new_key = wallet->deterministic_insert ((uint32_t)index, generate_work);
 			}
 		}
 		else
Code formatting differs from expected - please run ci/clang-format-all.sh
```

File: nano/node/rpc.cpp [hunk 1]
```diff
@@ -427,14 +427,35 @@ void nano::rpc_handler::account_create ()
...
 		const bool generate_work = request.get<bool> ("work", true);
-		nano::account new_key (wallet->deterministic_insert (generate_work));
-		if (!new_key.is_zero ())
+		nano::account new_key;
+		auto index_text (request.get_optional<std::string> ("index"));
+		if (index_text.is_initialized())
 		{
-			response_l.put ("account", new_key.to_account ());
+			uint64_t index;
+			if (decode_unsigned (index_text.get(), index) || index > (uint64_t) std::numeric_limits<uint32_t>::max ())
```
[the remark this answers is not in the document]
Comment (guilhermelawless) [reply nano/node/rpc.cpp]: Old habits die hard. Done!

[the remark this answers is not in the document]
Comment (wezrule) [reply nano/node/rpc.cpp]: Not sure they died enough :) those casts are essentially equivalent. I mean using static_cast<> as per "More Effective C++ Item 2 - Prefer C++-style casts" (Scott Meyers).

**Comment by guilhermelawless** [comment]
@argakiig Should indices higher than `2^33-1` be allowed? When testing it gives back the correct account, even though `deterministic_key_special` is `uint32_t` and will never reach those values sequentially.

File: nano/node/rpc.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (guilhermelawless) [reply nano/node/rpc.cpp]: Thanks, changed 👍

File: nano/node/wallet.cpp [hunk 2]
```diff
@@ -75,6 +75,19 @@ nano::public_key nano::wallet_store::deterministic_insert (nano::transaction con
 	return result;
 }
 
+nano::public_key nano::wallet_store::deterministic_insert (nano::transaction const & transaction_a, uint32_t const index)
+{
+	auto curr_index (deterministic_index_get (transaction_a));
```
Comment (SergiySW) [inline nano/node/wallet.cpp]: Seems not used

Comment (guilhermelawless) [reply nano/node/wallet.cpp]: Indeed, those were testing leftovers. An overloaded function seemed better at the end. Same for the other one, both removed.

**Pull request merged**